src/js/components/form-elements/v2/DateFieldDateFns.jsx

```diff
@@ -0,0 +1,210 @@
+import React, { useRef } from 'react';
+
+import { format, parse } from 'date-fns';
+import * as locales from 'date-fns/locale';
+import PropTypes from 'prop-types';
+import DatePicker from 'react-datepicker-6';
+import { useSelector } from 'react-redux';
+import { getCurrentLocale } from 'selectors';
+
+import DateFieldInput from 'components/form-elements/v2/DateFieldInput';
+import componentType from 'consts/componentType';
+import { DateFormatDateFns, TimeFormat } from 'consts/timeFormat';
+import useFocusOnMatch from 'hooks/useFocusOnMatch';
+import useTranslate from 'hooks/useTranslate';
+import InputWrapper from 'wrappers/InputWrapper';
+import RootPortalWrapper from 'wrappers/RootPortalWrapper';
+
+import 'components/form-elements/DateFilter/DateFilter.scss';
+import 'components/form-elements/v2/style.scss';
+
+const DateFieldDateFns = ({
+  title,
+  required,
+  tooltip,
+  disabled,
+  errorMessage,
+  placeholder,
+  button,
+  className,
+  value,
+  onChange,
+  showTimeSelect,
+  hideErrorMessageWrapper,
+  customDateFormat,
+  onChangeRaw,
+  clearable,
+  wrapperClassName,
+  focusProps = {},
+  ...fieldProps
+}) => {
+  const translate = useTranslate();
+  const onClear = () => onChange(null);
+
+  const { locale: currentLocale } = useSelector((state) => ({
+    locale: getCurrentLocale(state),
+  }));
+
+  const getDateFormat = () => {
+    if (showTimeSelect) {
+      return customDateFormat
+        ? `${customDateFormat} ${TimeFormat.HH_MM_SS}`
+        : DateFormatDateFns.MMM_DD_YYYY_HH_MM_SS;
+    }
+    return customDateFormat || DateFormatDateFns.MMM_DD_YYYY;
+  };
+
+  const formatDate = (dateToFormat) => {
+    if (!dateToFormat) {
+      return null;
+    }
+
+    try {
+      return parse(
+        dateToFormat,
+        getDateFormat(),
+        new Date(),
+      );
+    } catch (e) {
+      return null;
+    }
+  };
+
+  const selectedDate = formatDate(value);
+  const highlightedDates = [selectedDate || new Date()];
+
+  const placeholderText = typeof placeholder === 'object'
+    ? translate(placeholder?.id, placeholder?.defaultMessage)
+    : placeholder;
+
+  const datePickerRef = useRef(null);
+
+  const onChangeHandler = (date) => {
+    if (!date) {
+      onChange(null);
+      return;
+    }
+
+    const formatted = format(date, getDateFormat());
+    onChange(formatted);
+  };
+
+  const dateFnsLocale = () => {
+    if (!currentLocale || ['en', 'ar'].includes(currentLocale)) {
```

Comment:
I'm just noticing this now but why do we need this part: `['en', 'ar'].includes(currentLocale)`? ar is arabic right? Not english. And if `currentLocale == 'en'` won't that still get resolved by `locales[currentLocale]`?